### DISCUSSION TITLE
Feature Request: add option to make tabs look like tabs

### DISCUSSION THREAD

**Discussion by sionescu** [ghostty-org/ghostty#3268] [Feature Requests, Ideas] [answered]
I'd like to make a tab visually connect to the terminal that it signifies, as well as to the adjacent tabs, instead of being a floating visual element.

**Comment by mitchellh** [comment]
Converted to a discussion. Issues are not for feature requests, only accepted planned features. This discussion can be used to better discuss with the community the interest in this and design what it'll look like.

**Comment by sionescu** [comment]
Basically, I'd like tabs to looke like Chrome or Konsole tabs, with clear borders between them even when they're inactive, as well as making the closing button always visible.

**Reply by SuperDuperDeou** [reply]
I think that borders and close buttons should be handled by different options.

For the close buttons, a `tab-close-button-visibility = {always,focused,hovered,never}` should to the trick.

There might be problems at the toolkit level, tho, idk: I don't have much experience with GTK, let alone Apple's toolkit.

**Comment by lishaduck** [comment]
You can get this, to some extent, with `macos-titlebar-style = tabs`:
<img width="808" alt="Screenshot 2024-12-28 at 5 18 40 PM" src="https://github.com/user-attachments/assets/512e5994-699c-4c43-8c2a-c265ebbe1137" />

**Comment by sionescu** [comment]
I'm using Linux, so `macos-titlebar-style` probably doesn't apply.

**Reply by lishaduck** [reply]
Oh, yep 🤦‍♂️

**Comment by sionescu** [comment]
Solved this by disabling libAdwaita integration using `-Dgtk-adwaita=false`.

**Comment by sionescu** [comment]
It would be nice to have the default tabs a bit wider when libAdwaita is disabled and `gtk-wide-tabs = false`, because tabs are now so narrow it's hard to select them without pressing the close button. Something like what Chrome does: start with wider tabs and narrow them only if the tab bar is filled up.

**Comment by 00-kat** [comment, marked as answer]
This could probably be done with custom CSS (see [`gtk-custom-css`]).

[`gtk-custom-css`]: https://ghostty.org/docs/config/reference#gtk-custom-css
